src/main/java/cam72cam/immersiverailroading/entity/EntityRidableRollingStock.java

```diff
@@ -110,12 +110,13 @@ public boolean processInitialInteract(EntityPlayer player, EnumHand hand) {
 		} else if (player.isRiding() && player.getRidingEntity().getPersistentID() == this.getPersistentID()) {
 			return false;
 		} else {
-			if (!this.world.isRemote) {
+			if (!this.world.isRemote && (getLockType() != 2 || getLockOwner().equals(player.getUniqueID())) || playerHasOp(player)) {
```

Comment:
I see this added check in a few places.  Add a function on the rolling stock:
hasPermission(player, isBreaking)

Remove all the copy pasted conditionals

Reply:
Also this change does not work.  The client will get False even when the server get's true.

Take a look at the original code and see that both the client and server side always get the same answer